### PR TITLE
Compiler: Make LF 1.13 the default output

### LINKED ISSUE #8591: LF 1.13
Ticket for the next stable LF version after LF 1.12

Previous LF version ticket: #8369

Task:

- [x]  BigNumeric 
- [x] preview version (in SDK 1.13) 
- [x] release version (in SDK 1.14)
- [ ] default compiler output (in SDK 1.15) #9907

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Fixes #8591

CHANGELOG_BEGIN

- [Compiler] Default compiler output LF version is now 1.13

CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs [hunk 1]
```diff
@@ -46,7 +46,7 @@ version1_13 = V1 $ PointStable 13
 
 -- | The DAML-LF version used by default.
 versionDefault :: Version
-versionDefault = version1_12
+versionDefault = version1_13
```
Comment (cocreature) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs:49]: Aren’t you doing 1 step too many here for 1.14? We had 1.13 in preview in SDK 1.13. In SDK 1.14 we now need to make it stable and then we switch the default in SDK 1.15. Or am I missing something here?

**Comment by cocreature** [comment]
As discussed: this should wait until after SDK 1.14.

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 2]
```diff
@@ -104,9 +104,9 @@ conformance_test(
         "@jdk11_nix//:bin/java",
     ],
     lf_versions = [
-        "stable",
-        # FIXME: add latest once canton-test-runner supports LF 1.13
-        #  "latest",
+        # FIXME: replace "legacy" by "stable", "latest" once canton-test-runner supports LF 1.13
```
Comment (oliverse-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:107]: So this should no longer be necessary with 1e009b4ac2c24d118351f1af1e9689972def1dea

**Review by cocreature — APPROVED** [review]
thanks!

File: compatibility/bazel_tools/testing.bzl [hunk 3]
```diff
@@ -502,9 +502,12 @@ def daml_lf_compatible(sdk_version, platform_version):
     ) or (
         # any post 1.10.0 platform supports any pre 1.12 SDK
         in_range(platform_version, {"start": "1.10.0-snapshot"}) and not in_range(sdk_version, {"start": "1.12.0-snapshot"})
+    ) or (
+        # any post 1.10.0 platform supports any pre 1.14 SDK
+        in_range(platform_version, {"start": "1.11.0-snapshot"}) and not in_range(sdk_version, {"start": "1.14.0-snapshot"})
     ) or (
         # any post 1.11.0 platform supports any SDK
```
Comment (remyhaemmerle-da) [inline compatibility/bazel_tools/testing.bzl:509]:
```suggestion
        # any post 1.14.0 platform supports any SDK
```

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you!

**Pull request merged**